soh/src/overlays/actors/ovl_Bg_Jya_Cobra/z_bg_jya_cobra.c

```diff
@@ -585,6 +585,11 @@ void BgJyaCobra_DrawShadow(BgJyaCobra* this, PlayState* play) {
         Math_Vec3f_Copy(&sp64, &this->dyna.actor.world.pos);
     }
 
+    // Fixes cobra statue shadow texture not updating when rotating it
+    if (this->actionFunc == func_80896ABC) {
+        gSPInvalidateTexCache(POLY_XLU_DISP++, COBRA_SHADOW_TEX_PTR(this));
+    }
+
```

Comment:
This is basically the main fix for the texture updating (in addition to the helper macro)